### PR TITLE
Responsive popover fix for Safari

### LINKED ISSUE #7753: Popover height resizing issue in Safari
## Issue summary

In Safari, when the content of the popover changes, the popover `Pane` fails to resize. This appears to be an issue with the `Popover` `Pane` not shrinking to fit the content when the content changes while the popover is open.

From investigating this issue, I can see the `handleMeasurement` method in `PositionedOverlay` is being called correctly regardless of the browser. This method calculates the height using the height of the overlay, which is incorrect when the content changes in Safari. It looks like there might be an circumstantial issue with using `flex-basis` of `0%` in this case, as changing `flex-basis` to be `auto` or `content` [here](https://github.com/Shopify/polaris/blob/%40shopify/polaris%4010.12.0/polaris-react/src/components/Popover/Popover.scss#L92) solves the issue. I'm looking into whether changing the flex basis here would impact any other use cases of the `Popover` component.

Note that this is not an issue when a `Popover` `Pane` with `fixed` as true is the first child of the popover.

## Expected behavior

When the content of the popover changes, the popover should resize to fit the content. This works as expected in Chrome and Firefox, but Safari shows the previous content's height.

## Actual behavior

Safari calculates the popover's height to be the previous content's height, rather than re-measuring the new height correctly.

## Steps to reproduce the problem

1. Render a `Popover` with some initial content
1. Render the `Popover` with new content smaller than the initial content

## Reduced test case

This issue can be reproduced using the following codesandbox. On Safari, the popover won't resize to fit the `LoadedContent`. The content is changed via a state update 500ms after the popover is opened

https://codesandbox.io/s/popover-loading-content-kyi76m?file=/App.js

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 10.12.0
- Browser: Safari
- Device: Macbook Pro (M1 chip)
- Operating System: macOS 13.0 (22A380)

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

### PR DESCRIPTION
**Pull request by highfieldjames**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris/issues/7753

Fixes a popover resizing issue in Safari. See [the issue](https://github.com/Shopify/polaris/issues/7753) for more context.

### WHAT is this pull request doing?

This PR changes `flex-basis` from `0%` to `auto` for the `Pane` component in the shorthand `flex` property.

A `flex-basis` of `content` would probably make more sense here, but it's [not currently supported by Safari](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis#browser_compatibility).

Interesting side note: the `flex-basis` initial property is `auto`, but the shorthand `flex` property defaults flex-basis to 0.

#### Before change (Safari)

https://user-images.githubusercontent.com/6731221/212118475-ec25d862-b2b4-476a-9630-48907a1f51fb.mp4

#### After change (Safari)

https://user-images.githubusercontent.com/6731221/212119076-8770ab55-0705-4516-a925-0513ff3de3db.mp4

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useCallback, useState} from 'react';
import {Button, Page, Popover} from '../src';

export function Playground() {
  return (
  const [popoverActive, setPopoverActive] = useState(true);
  const [loading, setLoading] = useState(false);

  const togglePopoverActive = useCallback(
    () => setPopoverActive((popoverActive) => !popoverActive),
    [],
  );

  const activator = (
    <Button
      onClick={() => {
        setLoading(true);
        togglePopoverActive();
        window.setTimeout(() => {
          setLoading(false);
        }, 500);
      }}
      disclosure
    >
      Show server data
    </Button>
  );

  return (
    <Page title="Playground">
        <div style={{height: '280px'}}>
          <Popover
            active={popoverActive}
            activator={activator}
            onClose={togglePopoverActive}
            ariaHaspopup={false}
            sectioned
          >
            {loading ? (
              <div style={{height: '200px'}}>
                <p>Loading...</p>
              </div>
            ) : (
              <div>
                <p>Small content from the server</p>
              </div>
            )}
          </Popover>
        </div>
      );
    </Page>
  );
}
```

</details>

For Shopifolk, [this Spin instance](https://admin.web.web-popover-resize-fix.james-highfield.eu.spin.dev/store/shop1/products/inventory?location_id=1) is available to test on `@shopify/web`

[On this Storybook story on Chromatic](https://5d559397bae39100201eedc1-shwylzqduo.chromatic.com/?path=/story/all-components-popover--with-loading-smaller-content), open the popover. After 500ms, the popover's content should change, and the popover should resize to fit the content.

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/kind-feet-fly.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Fixed Safari bug with resizing popover pane
```
Comment (chloerice) [inline .changeset/kind-feet-fly.md]:
```suggestion
Fixed `Popover.Pane` not resizing when its content changes in Safari
```

**Review by chloerice — APPROVED** [review]

**Review by chloerice — APPROVED** [review]
Thank you for fixing this @highfieldjames, this works great!! 🙌🏽  You can go ahead and ship with the a11y check failing, #7854 fixes it

File: polaris-react/src/components/Popover/Popover.stories.tsx [hunk 2]
```diff
@@ -542,6 +542,53 @@ export function WithSearchableListbox() {
   );
 }
 
+export function WithLoadingSmallerContent() {
```
Comment (LauraAubin) [inline polaris-react/src/components/Popover/Popover.stories.tsx:545]: Love that you're adding a new story! ❤️

**Review by LauraAubin — APPROVED** [review]

**Pull request merged**